src/components/list/ItemsList.tsx

```diff
@@ -69,22 +84,29 @@ export default class ItemsList<T extends ListItem> extends React.Component<Props
     );
   }
 
-  private onSelectItem(item: T) {
-    const { selectedIds } = this.state;
+  private onSelectItem(item: T): void {
+    const { selectedItems } = this.state;
+    const { onSelect, select } = this.props;
+    const callback = onSelect ? () => onSelect([...this.state.selectedItems.values()]) : null;
```

Comment:
Move the callback in the if statement (no need to call values() if the item is not selected.
Rename callback to something more meaningful like 'itemSelectedCallback'

Reply:
The callback is used in the if and the else